src/main/java/org/killbill/billing/payment/api/PaymentApi.java

```diff
@@ -646,4 +646,12 @@ public void setDefaultPaymentMethod(Account account, UUID paymentMethodId, Itera
     @RequiresPermissions(PAYMENT_METHOD_CAN_UPDATE)
     public List<PaymentMethod> refreshPaymentMethods(Account account, Iterable<PluginProperty> properties, CallContext context)
             throws PaymentApiException;
-}
+
+    /**
+     * @param transactionId the payment transaction id
+     * @param context         the call context
+     * @return the payment transaction
+     * @throws PaymentApiException
+     */
+    public Payment getPaymentByTransactionId(final UUID transactionId, final boolean withPluginInfo, final boolean withAttempts, final Iterable<PluginProperty> properties, final TenantContext context) throws PaymentApiException;
```

Comment:
Nit: move it next to the other payment-related methods above (maybe after `getPaymentByExternalKey`).